src/Views/CategoryView/categoryView.cpp

```diff
@@ -134,6 +201,20 @@ int CategoryView::HandleInput(int focus, const Input& input)
 		focus = 0;
 	}
 
+	if (_ime_search_view_result.status == IMEVIEW_STATUS_FINISHED) {
```

Comment:
This is an abstraction leak, can you move all the logic code corresponding to searching in the IME file?

Reply:
The only way to circumvent this is adding functions to IMEView á la `isFinished()` `isCanceled`. I think this only complicates things. But if you insist I'll do.